src/__test__/authorize.test.ts

```diff
@@ -101,3 +101,77 @@ describe('authorize - with secret as callback in options', () => {
     })
   })
 })
+
+describe('authorize - with onAuthentication callback in options', () => {
+  let token: string = ''
+  let tokenWrong: string = ''
+
+  beforeEach(async (done) => {
+    jest.setTimeout(15_000)
+    await fixtureStart(
+      async () => {
+        const response = await axios.post('http://localhost:9000/login')
+        token = response.data.token
+        const responseWrong = await axios.post('http://localhost:9000/login-wrong')
+        tokenWrong = responseWrong.data.token
+        done()
+      },
+      {
+        secret: secretCallback,
+        onAuthentication: decodedToken => {
+          if (decodedToken.checkField === true) {
```

Comment:
Why are you checking if `decodedToken.checkField === true` with `===`, if it's a boolean ?
You could just write `decodedToken.checkField`.

Reply:
The linter is screaming murder (`Unexpected any value in conditional. An explicit comparison or type cast is required  @typescript-eslint/strict-boolean-expressions`) and conventional changelog is preventing me from committing a unlinted file.

Reply:
Oh okay, yeah now I see why there is this lint error.
To fix this, we could create a new type for the `profile` variable, an interface that you can export just above the `Socket` interface (in the same file).
I'm talking about this, it is in `src/__test__/fixture/index.ts` :
```ts
const profile = {
  email: 'john@doe.com',
  id: 123,
  checkField: false
}
```
And then import the interface and assign it to the decodedToken in the `onAuthentication` function like this :
```ts
onAuthentication: (decodedToken: Profile) => {
  if (!decodedToken.checkField) {
    throw new Error('Check Field validation failed')
  }
  return {
    email: decodedToken.email
  }
}
```
Hopefully, I'm clear, once this has been fixed, we can finally merge this!

Reply:
This is done